src/binary-reader-ir.cc

```diff
@@ -1612,6 +1613,11 @@ Result BinaryReaderIR::ValidateDataSegmentSymbol(Index symbol_index) {
   return Result::Ok;
 }
 
+Result BinaryReaderIR::OnRelocCount(Index count, Index section_index) {
+  module_->current_reloc_section_ = static_cast<BinarySection>(section_index);
```

Comment:
Instead of a direct cast here I think you need to lookup the type of section at that index, simular to the objdump reader:

```
reloc_section_ = section_types_[section_index];
```

Reply:
Never mind - I realized that I was wrong here.  I changed the code to mimic what objdump is doing.